config/src/integration-test/java/org/springframework/security/config/ldap/LdapProviderBeanDefinitionParserTests.java

```diff
@@ -0,0 +1,177 @@
+				+ "    </password-compare>"
+				+ "  </ldap-authentication-provider>"
+				+ "</authentication-manager>"
+				+ "<b:bean id='passwordEncoder' class='org.springframework.security.crypto.password.NoOpPasswordEncoder' factory-method='getInstance' />"
+		);
+
+		AuthenticationManager authenticationManager = appCtx.getBean(BeanIds.AUTHENTICATION_MANAGER, AuthenticationManager.class);
+		Authentication auth = authenticationManager.authenticate(new UsernamePasswordAuthenticationToken("ben", "ben"));
+
+		assertThat(auth).isNotNull();
+	}
+
+	// SEC-2472
+	@Test
+	public void supportsCryptoPasswordEncoder() {
+		appCtx = new InMemoryXmlApplicationContext("<ldap-server ldif='classpath:test-server.ldif'/>"
+				+ "<authentication-manager>"
+				+ "  <ldap-authentication-provider user-dn-pattern='uid={0},ou=people'>"
+				+ "    <password-compare>"
+				+ "      <password-encoder ref='pe' />"
+				+ "    </password-compare>"
+				+ "  </ldap-authentication-provider>"
+				+ "</authentication-manager>"
+				+ "<b:bean id='pe' class='org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder' />"
+		);
+
+		AuthenticationManager authenticationManager = appCtx.getBean(BeanIds.AUTHENTICATION_MANAGER, AuthenticationManager.class);
+		Authentication auth = authenticationManager.authenticate(new UsernamePasswordAuthenticationToken("bcrypt", "password"));
+
+		assertThat(auth).isNotNull();
+	}
+
+	@Test
+	public void inetOrgContextMapperIsSupported() {
+		appCtx = new InMemoryXmlApplicationContext("<ldap-server url='ldap://127.0.0.1:343/dc=springframework,dc=org'/>"
+				+ "<authentication-manager>"
+				+ "  <ldap-authentication-provider user-details-class='inetOrgPerson' />"
+				+ "</authentication-manager>"
+		);
+
+		ProviderManager providerManager = appCtx.getBean(BeanIds.AUTHENTICATION_MANAGER, ProviderManager.class);
+		assertThat(providerManager.getProviders()).hasSize(1);
+		assertThat(providerManager.getProviders())
+				.extracting("userDetailsContextMapper")
+				.allSatisfy(contextMapper -> assertThat(contextMapper).isInstanceOf(InetOrgPersonContextMapper.class));
+	}
+
+	@Test
+	public void ldapAuthenticationProviderWorksWithPlaceholders() {
+		System.setProperty("udp", "people");
+		System.setProperty("gsf", "member");
+		appCtx = new InMemoryXmlApplicationContext("<ldap-server />"
+				+ "<authentication-manager>"
+				+ "  <ldap-authentication-provider user-dn-pattern='uid={0},ou=${udp}' group-search-filter='${gsf}={0}' />"
+				+ "</authentication-manager>"
+				+ "<b:bean id='org.springframework.beans.factory.config.PropertyPlaceholderConfigurer' class='org.springframework.beans.factory.config.PropertyPlaceholderConfigurer' />"
+		);
+
+		ProviderManager providerManager = appCtx.getBean(BeanIds.AUTHENTICATION_MANAGER, ProviderManager.class);
+		assertThat(providerManager.getProviders()).hasSize(1);
+
+		AuthenticationProvider authenticationProvider = providerManager.getProviders().get(0);
+		assertThat(authenticationProvider)
```

Comment:
As of assertj 3.13 and higher `extracting` no longer returns a list, so this code does not compile.
For more information you can see the breaking changes section of the [assertj release notes ](https://assertj.github.io/doc/#assertj-core-3-13-0-release-notes).
On master we are using version 3.14.0.

Reply:
I rebased master and squashed the previous commits, and fixed the compilations errors.